nexus/src/app/sagas/instance_create.rs

```diff
@@ -281,36 +269,51 @@ impl NexusSaga for SagaInstanceCreate {
             ));
             subsaga_append(
                 "external_ip",
-                subsaga_builder,
+                subsaga_builder.build()?,
                 &mut builder,
                 repeat_params,
                 i,
             )?;
         }
 
-        // See the comment above where we add nodes for creating NICs.  We use
-        // the same pattern here.
-        for i in 0..(MAX_DISKS_PER_INSTANCE as usize) {
-            let disk_params =
-                DiskParams { saga_params: params.clone(), which: i };
+        // Create any specified disks
+        for (i, disk) in params.create_params.disks.iter().enumerate() {
+            if let InstanceDiskAttachment::Create(create_disk) = disk {
+                let subsaga_name =
+                    SagaName::new(&format!("instance-create-disk-{i}"));
+                let subsaga_builder = DagBuilder::new(subsaga_name);
+                subsaga_append(
+                    "create_disk",
+                    SagaDiskCreate::make_saga_dag(
+                        &disk_create::Params {
+                            serialized_authn: params.serialized_authn.clone(),
+                            project_id: params.project_id,
+                            create_params: create_disk.clone(),
+                        },
+                        subsaga_builder,
+                    )?,
+                    &mut builder,
+                    create_disk,
+                    i,
+                )?;
```

Comment:
The thing I'm most concerned about here is the `create_disk` being passed into subsaga_append. I don't know if that's the right params. I suspect what I'm passing into `make_saga_dag` is actually the right params...